packages/components/src/SegmentedControl/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useId, useMemo } from "react";
+import { useSegmentedControl } from "./SegmentedControlProvider";
+
+interface SegmentedControlOptionProps extends PropsWithChildren {
+  /**
+   * The value of the option.
+   */
+  readonly value: string | number;
+  /**
+   * The label to appear beside the radio button.
+   */
+  readonly checked?: boolean;
```

Comment:
This seems like an odd behaviour. Seems like checked should always come from the provider rather than enabling a situation where the top level value can reflect a different checked state than the option.

Reply:
Removed the `checked` prop here bfac31d85bc847ce0fc080890cc9818bea9f5838
Seems like selection state is already managed using `selectedOption` from `SegmentedControlContext`